src/mono/wasm/Wasm.Build.Tests/InvariantTimezoneTests.cs

```diff
@@ -0,0 +1,97 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using Xunit;
+using Xunit.Abstractions;
+
+#nullable enable
+
+namespace Wasm.Build.Tests
+{
+    public class InvariantTimezoneTests : BuildTestBase
+    {
+        public InvariantTimezoneTests(ITestOutputHelper output, SharedBuildPerTestClassFixture buildContext)
+            : base(output, buildContext)
+        {
+        }
+
+        public static IEnumerable<object?[]> InvariantTimezoneTestData(bool aot, RunHost host)
+            => ConfigWithAOTData(aot)
+                .Multiply(
+                    new object?[] { null },
+                    new object?[] { false },
+                    new object?[] { true })
+                .WithRunHosts(host)
+                .UnwrapItemsAsArrays();
+
+        [Theory]
+        [MemberData(nameof(InvariantTimezoneTestData), parameters: new object[] { /*aot*/ false, RunHost.All })]
+        [MemberData(nameof(InvariantTimezoneTestData), parameters: new object[] { /*aot*/ true, RunHost.All })]
+        public void AOT_InvariantTimezone(BuildArgs buildArgs, bool? invariantTimezone, RunHost host, string id)
+            => TestInvariantTimezone(buildArgs, invariantTimezone, host, id);
+
+        [Theory]
+        [MemberData(nameof(InvariantTimezoneTestData), parameters: new object[] { /*aot*/ false, RunHost.All })]
+        public void RelinkingWithoutAOT(BuildArgs buildArgs, bool? invariantTimezone, RunHost host, string id)
+            => TestInvariantTimezone(buildArgs, invariantTimezone, host, id,
+                                            extraProperties: "<WasmBuildNative>true</WasmBuildNative>",
+                                            dotnetWasmFromRuntimePack: false);
+
+        private void TestInvariantTimezone(BuildArgs buildArgs, bool? invariantTimezone,
+                                                        RunHost host, string id, string extraProperties="", bool? dotnetWasmFromRuntimePack=null)
+        {
+            string projectName = $"invariant_{invariantTimezone?.ToString() ?? "unset"}";
+            if (invariantTimezone != null)
+                extraProperties = $"{extraProperties}<InvariantTimezone>{invariantTimezone}</InvariantTimezone>";
+
+            buildArgs = buildArgs with { ProjectName = projectName };
+            buildArgs = ExpandBuildArgs(buildArgs, extraProperties);
+
+            if (dotnetWasmFromRuntimePack == null)
+                dotnetWasmFromRuntimePack = !(buildArgs.AOT || buildArgs.Config == "Release");
+
+            string programText = @"
```

Comment:
Please, put the source to `src\mono\wasm\testassets`

Reply:
This code seems specific to this test, so it's fine to have it contained here itself.